mne/viz/evoked.py

```diff
@@ -972,17 +980,27 @@ def _plot_evoked_white(evoked, noise_cov, scalings=None, rank=None, show=True):
             for ch_type, this_picks in picks_list2:
                 this_info = pick_info(evoked.info, this_picks)
                 idx = np.ix_(this_picks, this_picks)
+                this_estimated_rank = _estimate_rank_meeg_cov(
+                    C[idx], this_info, scalings)
+                expected_rank = len(this_picks)
+                expected_rank_reduction = 0
                 if has_meg and has_sss and ch_type in ('meg', 'grad', 'mag'):
-                    logger.info('Reading SSS rank from proc info')
-                    this_rank = get_rank_sss(evoked)
-                    logger.info('Looking for SSP vectors')
-                    n_ssp = sum(pp['active'] for pp in cov['projs'])
-                    this_rank -= n_ssp
-                    logger.info('final rank (%s) = %i' % (ch_type, this_rank))
-                else:
-                    this_rank = _estimate_rank_meeg_cov(C[idx], this_info,
-                                                        scalings)
-                rank_[ch_type] = this_rank
+                    sss_rank = _get_rank_sss(evoked)
+                    expected_rank_reduction += (expected_rank - sss_rank)
+                n_ssp = sum(pp['active'] for pp in cov['projs'] if
+                            _match_proj_type(pp, this_info['ch_names']))
+                expected_rank_reduction += n_ssp
+                expected_rank -= expected_rank_reduction
+                if this_estimated_rank != expected_rank:
+                    msg = (
+                        'For (%s) the expected and estimated rank diverge '
+                        '(%i VS %i). \nThis may lead to surprising reults. '
+                        '\nPlease consider using the `rank` parameter to '
+                        'manually specify the spatial degrees of freedom.'
+                    )
+                    msg %= (ch_type, expected_rank, this_estimated_rank)
+                    logger.warning(msg)
+                rank_[ch_type] = this_estimated_rank
```

Comment:
@Eric89GXL I propose a more liberal test here, let's always get informed when the rank is mismatched.